### ISSUE TITLE
Image is being uploaded with -90 degrees

### ISSUE THREAD

**Issue by pranavpari** [firebase/firebase-ios-sdk#3641]
most of the images I am trying to upload are being uploaded with -90 degrees

**Comment by ryanwilson** [comment]
Hi @pranavpari, a few questions:

- Which SDK are you talking about? Storage, MLKit, something else?
- Can you give a code sample along with example image so we can reproduce?
- Which version of Firebase are you using?

Thanks!

**Comment by pranavpari** [comment]
I am talking about FirebaseStorage. I am using 3.4.0.
My code sample
 
```
func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
        if let selectedImage = info[UIImagePickerController.InfoKey.originalImage] as? UIImage {
            imageData = selectedImage.pngData()
            self.uploadFile(data: imageData)
        }
        dismiss(animated: true, completion: nil)
    }

func uploadFile(data:Data){
        let uploadTask = riversRef.putData(data, metadata: nil) { (metadata, error) in
            guard let _ = metadata else {
                return
            }
            
            riversRef.downloadURL { (url, error) in
                guard let downloadURL = url else {
                    return
                }
        }
    }
```

**Comment by schmidt-sebastian** [comment]
@pranavpari Thanks for filing this issue. This sounds like you are running into a problem with iOS itself. The iOS camera records the rotation of a picture in EXIFF data, rather than just rotating the picture data. A fair number of image processing tools do not interpret this correctly, and hence display images from iOS rotated. See https://www.geek.com/apple/why-iphone-pictures-flip-when-emailed-and-how-to-fix-it-1602306/ for a short explanation.

Since we are not manipulating data in Storage but rather just passing it through, I am closing this for now.
